app/models/Ticket.js

```diff
@@ -0,0 +1,102 @@
+const mongoose = require('mongoose')
+const validator = require('validator')
+const Schema = mongoose.Schema
+const commentSchema = require('./Comment').schema
+
+const ticketSchema = new Schema({
+  title: {
+    type: String,
+    trim: true,
+    required: true
+  },
+  createdBy: {
+    type: Schema.Types.ObjectId,
+    ref: 'User'
+  },
+  status: {
+    type: String,
+    enum: ['Open', 'Closed', 'Pending', 'Closed'],
+    default: 'Open',
+    required: true
+  },
+  content: {
+    shortDescription: {
+      type: String,
+      required: true,
+      trim: true,
+      minlength: 5,
+      validate (shortDescription) {
+        if (validator.isEmpty(shortDescription)) {
+          throw new Error('Short description is required!')
+        }
+        if (!validator.isLength(shortDescription, { min: 10 })) {
+          throw new Error('Short description should be min 5 characters long!')
+        }
+      }
+    },
+    longDescription: {
+      type: String,
+      trim: true,
+      minlength: 10,
+      validate (longDescription) {
+        if (validator.isEmpty(longDescription)) {
+          throw new Error('Long description is required!')
+        }
+        if (!validator.isLength(longDescription, { min: 10 })) {
+          throw new Error('Long description should be min 10 characters long!')
+        }
+      }
+    }
+  },
+  tags: [
+    {
+      type: String,
+      trim: true,
+      validate: (value) => {
+        if (!validator.isLength(value, { min: 0, max: 20 })) {
+          throw new Error('Tags should have between 0 to 20 characters')
+        }
+      }
+    }
+  ],
+  history: [
+    {
+      title: {
+        type: String,
+        trim: true
+      },
+      content: {
+        shortDescription: {
+          type: String,
+          trim: true
+        },
+        longDescription: {
+          type: String,
+          trim: true
+        }
+      },
+      editedAt: {
+        type: Date,
+        required: true,
+        default: Date.now()
+      },
+      editedBy: {
+        type: Schema.Types.ObjectId,
+        ref: 'User'
+      }
+    }
+  ],
+  comments: [commentSchema], // mongoose subdocument
```

Comment:
I think if I use refs then it would have the document in the common Comments collection and I guess we would not want comments from various sources to be stored at the same place, I want the array elements to follow the commentSchema but stored inside our ticket document. Please do let me know if I am wrong on this? Thanks.
By the way, I ended up creating a new schema for the comments in the ticket system as I needed some different properties, is there any elegant way to extend or override existing schemas? Thanks again. Curious to learn.